frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -1879,7 +1879,7 @@ mod tests {
 			assert!(MultiPhase::snapshot().is_some());
 			assert_eq!(MultiPhase::round(), 1);
 
-			roll_to(25);
+			roll_to_unsigned();
```

Comment:
I think if we are using this pattern then we should not use the exact block numbers explicitly in the code anymore. What I mean is, the test should read something like this
```
roll_to_signed();
some_assertions();

roll_by(9); // instead of roll_to(24), go forward by 9 blocks
assertions_phase_still_signed();

roll_to_unsigned();
some_assertions();
```

My reasoning for this is, if we still have to keep track of block numbers, it defeats the purpose of using the pattern `roll_to_event()`

Since we are actively moving forward till a certain phase is reached, we probably don't need this assertions at block 24 anyway. If there is a logical assertion needed here, may be it could be done a bit differently. For example, `block_unsigned_phase - block_signed_phase == 10`

Reply:
cc @kianenigma

Reply:
The goal is to have as much "separation of concerns" as possible. There is one tests somewhere in there that checks `roll_to(x)` manually (called `phase_rotation_works`). All other tests should spend the minimum amount of mental effort to deal with phase related stuff, so I would stick to `roll_to_signed` and `roll_to_unsigned` for the sake of simplicity.